### PR TITLE
graphql-java-support: Bump graphql-java to v15

### PR DESCRIPTION
**Pull request by sachindshinde**
This PR is an iteration of the changes suggested in #67 (many thanks to the authors and commenters).

Summary of Changes:
- Bump `graphql-java` to v15 in the `pom.xml`
- Remove `enforceSchemaDirectives(true)` from `Federation.java`
  - Reason: This option was removed in graphql-java/graphql-java#1773 to facilitate spec compliance, and now is always `true` (thankfully we've already done the legwork to not have to worry about this)
- Remove `includeExtendedScalarTypes(true)` from `SchemaTransformer.java`
  - Reason: This option was removed in graphql-java/graphql-java#1780 to facilitate spec compliance, and now is always `true` (what were previously known as "extended scalars" in `graphql-java` are now just regular custom scalars)
- Add a new standard GraphQL directive `@specifiedBy` to standard directive lists
  - Reason: This directive was added to `graphql-java` in graphql-java/graphql-java#1772 with the spec addition still pending at graphql/graphql-spec#635
  - Note that it's unknown how gateway currently responds to the presence of this directive definition in the federation SDL, but it's assumed that it will eventually behave the same as it does with other standard directive definitions if it doesn't already do so. (This may introduce an issue in the interim that a user wanting to use `@specifiedBy` with federation will not be able to do so, since federation SDL won't expose the directive definition and the directive definitions the gateway introduces may not have it.)
- Update `FederationSdlPrinter` to use `graphql-java` v15's `SchemaPrinter`
  - Reason: As described in #53 , we use our own printer to avoid printing federation directive definitions and federation type definitions in SDL.
  - Note that the bugfix that we backported in #52 is no longer needed with graphql-java v15.

Note that this is a backwards incompatible change due to backwards incompatible changes in `graphql-java` v15's `SchemaPrinter`.

It'll be easier in this PR to go commit-by-commit. For `FederationSdlPrinter`, it'll help to compare `FederationSdlPrinter.java` in the class-copy-and-rename commit to the v15 `SchemaPrinter`, located at https://raw.githubusercontent.com/graphql-java/graphql-java/v15.0/src/main/java/graphql/schema/idl/SchemaPrinter.java . For the `FederationSdlPrinter` changes that fix the issue in #53 , it'll help to compare against the respective changes in that PR.

### CODE REVIEW

**Comment by sachindshinde** [comment]
Note that we should test this in our own Apollo infra before releasing

File: .idea/misc.xml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
   <component name="ExternalStorageConfigurationManager" enabled="true" />
+  <component name="JavaScriptSettings">
```
Comment (pcarrier) [inline .idea/misc.xml:4]: ?

File: graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/FederationSdlPrinter.java [hunk 2]
```diff
@@ -462,17 +468,36 @@ private void printFieldDefinitions(PrintWriter out, Comparator<? super GraphQLSc
...
             if (shouldPrintAsAst(type.getDefinition())) {
                 printAsAst(out, type.getDefinition(), type.getExtensionDefinitions());
             } else {
                 printComments(out, type, "");
-                out.format("interface %s%s", type.getName(), directivesString(GraphQLInterfaceType.class, type.getDirectives()));
+                if (type.getInterfaces().isEmpty()) {
+                    out.format("interface %s%s", type.getName(), directivesString(GraphQLInterfaceType.class, type.getDirectives()));
+                } else {
+
+                    GraphqlTypeComparatorEnvironment environment = GraphqlTypeComparatorEnvironment.newEnvironment()
```
Comment (pcarrier) [inline graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/FederationSdlPrinter.java:479]: Looks static

File: .idea/misc.xml (hunk 1 above)
Comment (sachindshinde) [reply .idea/misc.xml:4]: My IntelliJ's been adding that in for a while, no clue why it keeps doing that (maybe it's because of plugins?).

**Pull request merged**